### PR TITLE
[SIP-5] Refactor and repair partition

### PR DESCRIPTION
**Pull request by kristw**
- Extract `slice` and `formData` according to [SIP-5]
- Fix small issues and syntax inside heatmap.
- Fix broken tooltip: Add background, reorder columns into readable format. add padding between columns

@williaster @conglei 

Before
icicle_test-2

After
icicle_test

### CODE REVIEW

File: superset/assets/src/visualizations/partition.js [hunk 1]
```diff
@@ -221,26 +250,52 @@ function partitionVis(slice, payload) {
...
         });
+      } else {
+        t += (
+          '<thead><tr><td colspan="3">' +
+            `<strong class='x-value'>${getCategory(d.depth)}</strong>` +
+            '</td></tr></thead><tbody>'
+        );
+        t += (
+          '<tr class="emph">' +
+            '<td class="legend-color-guide" style="opacity: 0.75">' +
```
Comment (williaster) [inline superset/assets/src/visualizations/partition.js]: do styles for `.legend-color-guide` and `.x-value` come from another style sheet? or do they do anything?

**Review by williaster — APPROVED** [review]
couple comments, looks solid overall! 🎈

File: superset/assets/src/visualizations/partition.js [hunk 2]
```diff
@@ -324,10 +345,78 @@ function partitionVis(slice, payload) {
...
+    width: slice.width(),
+    height: slice.height(),
+    colorScheme,
+    dateTimeFormat,
+    equalDateSize,
+    groupBy,
+    logScale,
+    metrics,
+    numberFormat,
+    partitionLimit: partitionLimit ? parseInt(partitionLimit, 10) : partitionLimit,
```
[the remark this answers is not in the document]
Comment (kristw) [reply superset/assets/src/visualizations/partition.js]: 👍

**Comment by kristw** [comment]
I addressed the comments above.
- Remove commented code
- Rename `kx` `ky`
- Remove legacy className from nvd3 tooltip
- Remove `verboseMap` from prop.

**Pull request merged**